standalone/test/include/psen_scan_v2_standalone/util/matchers_and_actions.h

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2021 Pilz GmbH & Co. KG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+#ifndef PSEN_SCAN_V2_STANDALONE_TEST_MATCHERS_AND_ACTIONS_H
+#define PSEN_SCAN_V2_STANDALONE_TEST_MATCHERS_AND_ACTIONS_H
+
+#include "psen_scan_v2_standalone/util/timestamp.h"
+
+namespace psen_scan_v2_standalone_test
+{
+using namespace ::testing;
+using namespace ::psen_scan_v2_standalone;
+
+ACTION_P(OpenBarrier, barrier)
+{
+  barrier->release();
+}
+
+MATCHER_P(PointwiseDoubleEq, vec, "")
+{
+  return std::equal(vec.begin(), vec.end(), arg.begin(), arg.end(), [](const double& a, const double& b) {
+    return Matches(DoubleEq(b))(a);
+  });
+}
+
+MATCHER_P(ScanDataEqual, scan, "")
+{
+  return arg.getScanCounter() == scan.getScanCounter() && arg.getScanResolution() == scan.getScanResolution() &&
```

Comment:
See http://google.github.io/googletest/gmock_cook_book.html#CombiningMatchers and considere using it. I assume on a failure one wants to know what **exactly** failed.

Reply:
I only moved this code. Would suggest to do any changes regarding the logic in a separate PR